### ISSUE TITLE
docker cuda error

### ISSUE THREAD

**Issue by fanlu** [SeanNaren/deepspeech.pytorch#341]
when I used --cuda with docker, the loss value was always 0,That was not happened in cpu version.
the log of gpu version
```
root@7096c8ab06ef:/workspace/deepspeech.pytorch# python train.py --cuda  --train-manifest /workspace/data/libri_train_manifest.csv --val-manifest /workspace/data/libri_val_manifest.csv
DeepSpeech(
  (conv): MaskConv(
    (seq_module): Sequential(
      (0): Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5))
      (1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): Hardtanh(min_val=0, max_val=20, inplace)
      (3): Conv2d(32, 32, kernel_size=(21, 11), stride=(2, 1), padding=(10, 5))
      (4): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (5): Hardtanh(min_val=0, max_val=20, inplace)
    )
  )
  (rnns): Sequential(
    (0): BatchRNN(
      (rnn): GRU(1312, 800, bidirectional=True)
    )
    (1): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (2): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (3): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (4): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
  )
  (fc): Sequential(
    (0): SequenceWise (
    Sequential(
      (0): BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (1): Linear(in_features=800, out_features=29, bias=False)
    ))
  )
  (inference_softmax): InferenceBatchSoftmax()
)
Number of parameters: 41187968
Epoch: [1][1/1017]      Time 0.630 (0.630)      Data 0.298 (0.298)      Loss 0.0000 (0.0000)
Epoch: [1][2/1017]      Time 0.262 (0.446)      Data 0.005 (0.152)      Loss 0.0000 (0.0000)
Epoch: [1][3/1017]      Time 0.264 (0.386)      Data 0.004 (0.103)      Loss 0.0000 (0.0000)
Epoch: [1][4/1017]      Time 0.272 (0.357)      Data 0.004 (0.078)      Loss 0.0000 (0.0000)
Epoch: [1][5/1017]      Time 0.273 (0.340)      Data 0.003 (0.063)      Loss 0.0000 (0.0000)
Epoch: [1][6/1017]      Time 0.269 (0.328)      Data 0.003 (0.053)      Loss 0.0000 (0.0000)
Epoch: [1][7/1017]      Time 0.276 (0.321)      Data 0.003 (0.046)      Loss 0.0000 (0.0000)
```
the log of cpu version
```
^oot@7096c8ab06ef:/workspace/deepspeech.pytorch# python train.py  --train-manifest /workspace/data/libri_train_manifest.csv --val-manifest /workspace/data/libri_val_manifest.csv       
DeepSpeech(
  (conv): MaskConv(
    (seq_module): Sequential(
      (0): Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 2), padding=(20, 5))
      (1): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (2): Hardtanh(min_val=0, max_val=20, inplace)
      (3): Conv2d(32, 32, kernel_size=(21, 11), stride=(2, 1), padding=(10, 5))
      (4): BatchNorm2d(32, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (5): Hardtanh(min_val=0, max_val=20, inplace)
    )
  )
  (rnns): Sequential(
    (0): BatchRNN(
      (rnn): GRU(1312, 800, bidirectional=True)
    )
    (1): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (2): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (3): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
    (4): BatchRNN(
      (batch_norm): SequenceWise (
      BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True))
      (rnn): GRU(800, 800, bidirectional=True)
    )
  )
  (fc): Sequential(
    (0): SequenceWise (
    Sequential(
      (0): BatchNorm1d(800, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)
      (1): Linear(in_features=800, out_features=29, bias=False)
    ))
  )
  (inference_softmax): InferenceBatchSoftmax()
)
Number of parameters: 41187968
Epoch: [1][1/1017]      Time 13.389 (13.389)    Data 0.157 (0.157)      Loss 237.1903 (237.1903)
Epoch: [1][2/1017]      Time 9.779 (11.584)     Data 0.032 (0.094)      Loss 241.4673 (239.3288)
Epoch: [1][3/1017]      Time 10.254 (11.140)    Data 0.054 (0.081)      Loss 193.7481 (224.1352)
Epoch: [1][4/1017]      Time 10.397 (10.955)    Data 0.054 (0.074)      Loss 161.4310 (208.4592)
```

**Comment by fanlu** [comment]
I use docker to build Dockerfile in project, not use nvidia-docker to build. Should I use nvidia-docker to rebuild an image instead of docker?

**Comment by ryanleary** [comment]
Yes, probably. warp-ctc is going to compile against cuda and it may look at what compute capability your card has when doing the build. That would be a good first step to try at least (using nvidia-docker and building/running on the same machine w/ same card).
